shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -162,6 +163,16 @@ static bool OnAcquireExternalTexture(FlutterEngine* engine,
 
 #pragma mark -
 
+namespace {
+
+struct AotDataDeleter {
+  void operator()(FlutterEngineAOTData aot_data) { FlutterEngineCollectAOTData(aot_data); }
+};
+
+using UniqueAotDataPtr = std::unique_ptr<_FlutterEngineAOTData, AotDataDeleter>;
```

Comment:
This doesn't really feel like it's being used as a `unique_ptr`, but rather just as an RAII wrapper that does some cleanup on destruction. It doesn't look like this actually deletes `aot_data` after it runs the collect function on it.

See the comment below about passing a non-null data ptr to the create function since it's related to lifetime of this struct.